oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/api/client.js

```diff
@@ -78,43 +81,108 @@ export class DepositApiClient {
  * API Client for deposits.
  */
 export class ApiClient extends DepositApiClient {
-  async _createResponse(axiosRequest) {
+  constructor(createUrl) {
+    super();
+    this.createUrl = createUrl;
+  }
+  _createResponse = async (axiosRequest) => {
     try {
       const response = await axiosRequest();
       const data = response.data || {};
       return data;
     } catch (error) {
-      const errorData = error.response.data;
-      return Promise.reject(errorData);
+      return Promise.reject(error);
     }
-  }
+  };
 
   /**
    * Calls the API to create a new draft.
    *
    * @param {object} draft - Serialized draft
    */
-  async createDraft(url, payload) {
-    return this._createResponse(() => this.axiosWithConfig.post(url, payload));
-  }
+  createDraft = async (draft) => {
+    return this._createResponse(() =>
+      this.axiosWithConfig.post(
+        this.createUrl,
```

Comment:
This is the only place createURl is needed, it would make sense to me to move it from instance to this function params

Reply:
@mirekys Ahoj Miro, firstly, thanks a lot for all the detailed comments you have made. I have already resolved most of the stuff, but for a few items, I have some problems. Initially, I was passing the createUrl as a param to createDraft method, but it turned out to be kind of problematic.

For example, the caller of createDraft, in case of NR-DOCS would be our depositActions class, that does not really have access to createUrl either. So this would mean I would kind of have to pass it from the button where I am calling the method from depositActions, to be passed down to the createDraft in client.js.

Maybe some compromise would be to add createUrl as a param to createDraft, but give it default value from the class constructor and in that way if someone chooses to use createDraft straight from the low level API class, they can provide the create URL themselves? Somehting like this:

`  constructor(createUrl = undefined) {
    super();
    this.createUrl = createUrl;
  }`


`  createDraft = async (draft, createUrl = this.createUrl) => {
    if (!createUrl)
      throw new Error(
        "You must either pass createUrl when initializing the OARepoDepositApiClient class or pass it to createDraft method., "
      );
    return this._createResponse(() =>
      this.axiosWithConfig.post(createUrl, draft)
    );
  };`